Nos/Models/NoteParser.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  NoteParser.swift
+//  Nos
+//
+//  Created by Martin Dutra on 17/4/23.
+//
+
+import CoreData
+import Foundation
+import RegexBuilder
+
+/// This struct encapsulates the algorithms that parse notes and the mentions inside the note.
+struct NoteParser {
+
+    /// Parses attributed text generated when composing a note and returns
+    /// the content and tags.
+    func parse(attributedText: AttributedString) -> (String, [[String]]) {
+        cleanLinks(in: attributedText)
+    }
+
+    /// Parses the content and tags stored in a note and returns an attributed text that can be used for displaying
+    /// the note in the UI.
+    func parse(content: String, tags: [[String]], context: NSManagedObjectContext) -> AttributedString {
+        // swiftlint:disable all
+        let regex = /(?:^|\s)#\[(?<index>\d+)\]|(?:^|\s)(?:nostr:)(?<npub>[-a-zA-Z0-9@:%._\+~#=]{2,256})/
```

Comment:
Do we need to include nprofile in this regex? I see you added support for it further below.

Reply:
I don't quite get your comment, parsing the nprofile in the regex seems central to me. Can you expand?

Reply:
Maybe the npub name of the capture group is misleading as it can capture both npub or nprofile strings. Was that the source of your confusion? I will change it to something else.

Reply:
Oh yeah, that did confuse me, but only because I don't understand regexes 😛